src/script/cpp_api/s_async.cpp

```diff
@@ -198,19 +310,22 @@ AsyncWorkerThread::~AsyncWorkerThread()
 /******************************************************************************/
 void* AsyncWorkerThread::run()
 {
-	lua_State *L = getStack();
-
-	try {
-		loadMod(getServer()->getBuiltinLuaPath() + DIR_DELIM + "init.lua",
-			BUILTIN_MOD_NAME);
-	} catch (const ModError &e) {
-		errorstream << "Execution of async base environment failed: "
-			<< e.what() << std::endl;
-		FATAL_ERROR("Execution of async base environment failed");
+	if (isErrored) {
+		sleep_ms(100); // Thread implementation bug workaround
```

Comment:
More information would be nice - what bug is this? On what platforms? Can this ever be removed?

Reply:
If a thread immediately exits this loop never ends:
https://github.com/minetest/minetest/blob/ec4a789b4f56d7918b6c964e59b9903af5e17fcd/src/threading/thread.cpp#L127-L128
I remember this being talked about somewhere.

Reply:
Now that I look at this, maybe the following works:
```diff
diff --git a/src/threading/thread.cpp b/src/threading/thread.cpp
index 5cfc60995..ef025ac1d 100644
--- a/src/threading/thread.cpp
+++ b/src/threading/thread.cpp
@@ -121,12 +121,12 @@ bool Thread::start()
                return false;
        }
 
-       // Allow spawned thread to continue
-       m_start_finished_mutex.unlock();
-
        while (!m_running)
                sleep_ms(1);
 
+       // Allow spawned thread to continue
+       m_start_finished_mutex.unlock();
+
        m_joinable = true;
 
        return true;
```

Reply:
Added a fixing commit.